### PR TITLE
[WIP] Autonity contract upgrade prototype

### PR DESCRIPTION
**Pull request by b00ris**

### CODE REVIEW

File: contracts/autonity/solidity/truffle-config.js [hunk 1]
```diff
@@ -42,11 +42,14 @@ module.exports = {
     // tab if you use this network and you must also set the `host`, `port` and `network_id`
     // options below to some value.
     //
-    // development: {
-    //  host: "127.0.0.1",     // Localhost (default: none)
-    //  port: 8545,            // Standard Ethereum port (default: none)
-    //  network_id: "*",       // Any network (default: none)
-    // },
+    development: {
```
Comment (Jason-Zhangxin-Chen) [inline contracts/autonity/solidity/truffle-config.js]: Hi Borris, by using this new configuraiton, which client shall truffle connects to for the contract binary deployment? I guess we need to setup a autonity client with same endpoint spec by this configuration, is it right?

**Comment by JekaMas** [comment]
@b00ris are you still working on the issue?
@yazzaoui or you are working on it?

**Comment by yazzaoui** [comment]
> @b00ris are you still working on the issue?
> @yazzaoui or you are working on it?

I'm working on it; should have the basic functionalities ready by tomorrow

**Comment by yazzaoui** [comment]
Haven't tested the code and it's far from being finished but the basic functionality should be implemented

File: contracts/autonity/solidity/contracts/Autonity.sol [hunk 2]
```diff
@@ -44,14 +44,24 @@ contract Autonity {
     }
 
     mapping (address => User) private users;
+    address[] private usersList;
```
Comment (aesmonty) [inline contracts/autonity/solidity/contracts/Autonity.sol]: I think there is a bug here. Deleted users are not deleted from `usersList` when they are removed by the system operator in `function removeUser`

Comment (Jason-Zhangxin-Chen) [reply contracts/autonity/solidity/contracts/Autonity.sol]: Yes, it was fix in my PR request from economic metrics branch

**Pull request closed without merging**